### ISSUE TITLE
suport for 

### ISSUE THREAD

**Issue by sstudent** [openshwprojects/OpenBK7231T_App#215]
Very good job! Could I please you to add support for Tuya Curtain Smart Switch with 
WB3S chip. Below I put pictures of the device.
![20220926_202556](https://user-images.githubusercontent.com/12023090/192356193-de30d8c8-3350-4e22-999e-204d575ad6bd.jpg)
![20220926_202655](https://user-images.githubusercontent.com/12023090/192356195-4785b0ac-3459-4013-969e-c4579e65d06f.jpg)
![20220926_202728](https://user-images.githubusercontent.com/12023090/192356197-ad12d30b-bb42-47c0-9be9-e1f65a2e0c5c.jpg)
![20220926_202737](https://user-images.githubusercontent.com/12023090/192356201-2f4bce2e-e00e-4b25-81ef-f847f4d77aad.jpg)
![20220926_202748](https://user-images.githubusercontent.com/12023090/192356203-f464c879-c649-4275-82a2-45b7a83d2c34.jpg)
Thank you

**Comment by DSchndr** [comment]
Do you have a logic analyzer / serial adapter?
Hook up rx from dongle to rx and later tx pad (gnd and vcc too) and monitor what ofw sends and receives 
(DO NOT CONNECT THE MAINS PART)

Product link would be nice :)

**Comment by DSchndr** [comment]
Pardon, did you use the rx/tx near the microcontroller (which I assumed are uart1) or did you solder to uart2 (log) on the module?

There should be data similar to this: https://tasmota.github.io/docs/TuyaMCU/#tuya-protocols

**Comment by sstudent** [comment]
Here is the answer on uart1 loged on realterm: 00 11 FF FF FF
And after push the batons:
00 11 FF FF FF CC 04 07 CC 04 0F CC 02 0B CC 02 0F CC 01 0D CC 01 0F CC 02
0B CC 02 0F CC 01 0D CC 01 0F
In the attached I send you logs from logic progam (Logic analyzer)

[0XFF.zip](https://github.com/openshwprojects/OpenBK7231T_App/files/9717781/0XFF.zip)

**Comment by sstudent** [comment]
Was it about these logs? Will it be possible to implement OpenBK on this device?

**Comment by DSchndr** [comment]
Probably, the logic analyzer logs seem to be empty, could you hook it up and save them in multiple files like "button up" 
You have to do something when you log them

**Comment by sstudent** [comment]
Here you have all logs
[logs.zip](https://github.com/openshwprojects/OpenBK7231T_App/files/9749452/logs.zip)

**Comment by DSchndr** [comment]
Sorry, since uni started I don't have a lot of time

Doesn't seem to be tuyamcu protocol, but seems to be something simple to implement, uart seems to run at 115200 baud

0x11 0xff 0xff 0xff -> Keepalive from mcu?
0xCC (static) 0x?? (button) 0x?? (state)

Do you have logs from changing states from the tuya app too?

**Comment by sstudent** [comment]
Here you have logs when I switch off and on the curtain from the smartlife app. 
Is sims to be empty.
[logs_smartlife.zip](https://github.com/openshwprojects/OpenBK7231T_App/files/9807246/logs_smartlife.zip)

**Comment by DSchndr** [comment]
Wierd, have you connected both UART lines to the Logic Analyzer?

**Comment by sstudent** [comment]
Only tx. Rx should be also conected?

**Comment by DSchndr** [comment]
Yes, since you need both the data that gets sent as well as received (on both sides)

**Comment by sstudent** [comment]
Nothing there on both Tx or Rx, when I turn on the curtain from smartlife app (everything works). Only when I push physically the button logs are ok. Can you make software for tests?

**Comment by DSchndr** [comment]
Need device to reverse engineer how it works and test, sorry

Cant do anything since I have no plan on how that thing works.
You could try to find out how the wifi module is connected to the rest and document it though
